File: scripts/verify_db.py
import urllib.request, json, ssl

ctx = ssl.create_default_context()
URL = 'https://ektbhxjpextknqouksnc.supabase.co'
KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVrdGJoeGpwZXh0a25xb3Vrc25jIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4MTE2NjY4NCwiZXhwIjoyMDk2NzQyNjg0fQ.cqtjYNhzfIF0mGNQggacrycOxuOKmjcgaJ8r31IgAa4'
HEADERS = {'apikey': KEY, 'Authorization': f'Bearer {KEY}', 'Content-Type': 'application/json'}

def get(path):
    req = urllib.request.Request(f'{URL}/rest/v1/{path}', headers=HEADERS)
    try:
        res = urllib.request.urlopen(req, context=ctx, timeout=8)
        return json.loads(res.read()), None
    except urllib.error.HTTPError as e:
        return None, f'{e.code}: {e.read().decode()[:100]}'

# 1. Table check
print('=== TABLE CHECK ===')
tables = ['admins','teams','events','media','notifications',
          'prayer_schedules','bible_quotes','organization',
          'settings','audit_logs','otp_sessions']
all_ok = True
for t in tables:
    data, err = get(f'{t}?limit=1&select=*')
    status = 'OK  ' if data is not None else 'FAIL'
    if data is None:
        all_ok = False
    print(f'  {status}  {t}' + (f'  -> {err}' if err else ''))

# 2. Teams
print('\n=== TEAMS SEEDED ===')
data, _ = get('teams?select=name,colour&order=name')
if data:
    for t in data:
        print(f'  {t["name"]}  ({t["colour"]})')
    print(f'  Total: {len(data)} teams (expected 9)')
    if len(data) != 9:
        print('  WARNING: Expected 9 teams!')

# 3. Bible quotes
print('\n=== BIBLE QUOTES ===')
data, _ = get('bible_quotes?select=reference&limit=10')
if data:
    for v in data:
        print(f'  {v["reference"]}')
    print(f'  Total: {len(data)} verses seeded')

# 4. Organization
print('\n=== ORGANIZATION ===')
data, _ = get('organization?select=name,tagline,founded_year')
if data:
    for o in data:
        print(f'  Name: {o["name"]}')
        print(f'  Tagline: {o["tagline"]}')
        print(f'  Founded: {o["founded_year"]}')
else:
    print('  MISSING - organization row not found!')

# 5. Settings
print('\n=== SETTINGS ===')
data, _ = get('settings?select=key,value')
if data:
    for s in data:
        print(f'  {s["key"]} = {s["value"]}')

# 6. Admins
print('\n=== ADMINS (your first admin) ===')
data, _ = get('admins?select=id,name,phone,role&limit=5')
if data:
    for a in data:
        print(f'  {a["name"]} | {a["phone"]} | {a["role"]}')
else:
    print('  NO ADMINS YET - add yourself to the admins table!')

print('\n=== SUMMARY ===')
print('All tables OK!' if all_ok else 'Some tables MISSING - re-run schema!')

File: scripts/verify_db.js
// Quick DB verification using existing Supabase client
const { createClient } = require('@supabase/supabase-js');

const supabase = createClient(
  'https://ektbhxjpextknqouksnc.supabase.co',
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVrdGJoeGpwZXh0a25xb3Vrc25jIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4MTE2NjY4NCwiZXhwIjoyMDk2NzQyNjg0fQ.cqtjYNhzfIF0mGNQggacrycOxuOKmjcgaJ8r31IgAa4'
);

async function verify() {
  const tables = [
    'admins','teams','events','media','notifications',
    'prayer_schedules','bible_quotes','organization',
    'settings','audit_logs','otp_sessions'
  ];

  console.log('=== TABLE CHECK ===');
  let allOk = true;
  for (const table of tables) {
    const { data, error } = await supabase.from(table).select('*').limit(1);
    const ok = !error;
    if (!ok) allOk = false;
    console.log(`  ${ok ? 'OK  ' : 'FAIL'} ${table}${error ? ' -> ' + error.message : ''}`);
  }

  console.log('\n=== TEAMS SEEDED ===');
  const { data: teams } = await supabase.from('teams').select('name,colour').order('name');
  (teams || []).forEach(t => console.log(`  ${t.name}  (${t.colour})`));
  console.log(`  Total: ${(teams||[]).length} teams (expected 9)`);

  console.log('\n=== BIBLE QUOTES ===');
  const { data: verses } = await supabase.from('bible_quotes').select('reference');
  (verses || []).forEach(v => console.log(`  ${v.reference}`));

  console.log('\n=== ORGANIZATION ===');
  const { data: org } = await supabase.from('organization').select('name,tagline,founded_year');
  if (org && org[0]) {
    console.log(`  Name: ${org[0].name}`);
    console.log(`  Tagline: ${org[0].tagline}`);
    console.log(`  Founded: ${org[0].founded_year}`);
  } else {
    console.log('  MISSING!');
  }

  console.log('\n=== SETTINGS ===');
  const { data: settings } = await supabase.from('settings').select('key,value');
  (settings || []).forEach(s => console.log(`  ${s.key} = ${s.value}`));

  console.log('\n=== ADMINS ===');
  const { data: admins } = await supabase.from('admins').select('name,phone,role').limit(5);
  if (admins && admins.length > 0) {
    admins.forEach(a => console.log(`  ${a.name} | ${a.phone} | ${a.role}`));
  } else {
    console.log('  NO ADMINS YET - you need to add yourself!');
  }

  console.log('\n=== SUMMARY ===');
  console.log(allOk ? '✅ All tables OK!' : '❌ Some tables MISSING - re-run schema!');
}

verify().catch(console.error);
